pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -83,19 +83,21 @@ def pre_optimizer_step(
                 f"native PyTorch amp and lbfgs are not compatible (optimizer {optimizer_idx})."
                 " To request, please file a Github issue in PyTorch and tag @mcarilli"
             )
-        lambda_closure()
```

Comment:
@yifuwang why did this PR change it so the closure is not executed `if not automatic_optimization`?

Reply:
I think it was because `training_step` was called directly (as opposed to through closure) for `manual_optimization` when this change was made: https://github.com/yifuwang/pytorch-lightning/blob/b4ef91ee835528c93e83186d1abecbb0ee129f98/pytorch_lightning/trainer/training_loop.py#L678

Reply:
I don't think that's right because here, `lambda_closure` would be a a custom closure passed to `optimizer.step(closure=...)` in the case of manual, not the full `training_step`.

Exploring this in #9288

Reply:
@carmocca I see. Maybe I misunderstood how it worked. Sorry for any issues this may have caused.